pilot/pkg/networking/plugin/httpmx/httpmx.go

```diff
@@ -0,0 +1,109 @@
+package httpmx
+
+import (
+	v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	wasm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/wasm/v3"
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	wasm_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/wasm/v3"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+)
+
+// Plugin implements Istio Telemetry HTTP metadata exchange
```

Comment:
Generally the metadata exchange plugin should cover both tcp and http, IMO we should not implement one for each

Reply:
yes, this is just a WIP PR for HTTP metadata exchange in order to gather some feedbacks from the developers, I will include both of them in the up coming commits and update the title of it